In [1]:
import requests

In [ ]:
url_ex = "https://api.census.gov/data/
2014/pep/natstprc?get=STNAME,POP&DATE_=7&for=state:*"
    
OR = '&for=state:01'

You can include up to 50 variables in a single API query and can make up to 500 queries per IP
address per day. More than 500 queries per IP address per day requires that you register for a
Census key. That key will be part of your data request URL string.


Every query must include a geography, and this API supports both FIPS and GNIS geography codes.

http://api.census.gov/data/2010/sf1?key=...&get=P0010001&for=state:06
https://tigerweb.geo.census.gov/arcgis/rest/services/Generalized_ACS2018?f=pjson

In [ ]:
base = "https://api.census.gov/data/"
year = "2018/"
 
acs_5yr_profiles = "acs/acs5/"
acs_5yr_subject_tables = "acs/acs5/subject"
acs1?get=NAME,B02015_009E,B02015_009M&for=state:*

In [ ]:
datasetname_acronym =acs_5yr_subject_tables


In [ ]:
tigerbase = "https://tigerweb.geo.census.gov/arcgis/rest/services/"
tracts = "Census2010/Tracts_Blocks/MapServer?"
&text=Oregon
returnZ = "&returnZ=true"
filetype = "f=geojson" 
#types are json, kmz, geojson

http://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer/3/query?where=&text=&objectIds=&time=&geometry=&geometryType=esriGeometryEnvelope&inSR=&spatialRel=esriSpatialRelIntersects&relationParam=&outFields=&returnGeometry=true&maxAllowableOffset=&outSR=&returnIdsOnly=false&returnCountOnly=false&orderByFields=&groupByFieldsForStatistics=sub_region&outStatistics=[{%22statisticType%22:%22sum%22,%22onStatisticField%22:%22pop2007%22,%22outStatisticFieldName%22:%22Population_2007%22},{%22statisticType%22:%22avg%22,%22onStatisticField%22:%22AVE_FAM_SZ%22,%22outStatisticFieldName%22:%22Average_Family_Size%22}]&returnZ=false&returnM=false&gdbVersion=&f=pjson

In [41]:
import requests
from pprint import pprint
import pandas as pd
import time

In [3]:
url = "https://api.census.gov/data/2018/acs/acs5/subject"

In [7]:
request = requests.get(url).json()

In [10]:
pprint(request)

{'@context': 'https://project-open-data.cio.gov/v1.1/schema/catalog.jsonld',
 '@id': 'https://api.census.gov/data/2018/acs/acs5/subject.json',
 '@type': 'dcat:Catalog',
 'conformsTo': 'https://project-open-data.cio.gov/v1.1/schema',
 'dataset': [{'@type': 'dcat:Dataset',
              'c_dataset': ['acs', 'acs5', 'subject'],
              'c_documentationLink': 'https://www.census.gov/developer/',
              'c_examplesLink': 'https://api.census.gov/data/2018/acs/acs5/subject/examples.json',
              'c_geographyLink': 'https://api.census.gov/data/2018/acs/acs5/subject/geography.json',
              'c_groupsLink': 'https://api.census.gov/data/2018/acs/acs5/subject/groups.json',
              'c_isAggregate': True,
              'c_isAvailable': True,
              'c_isCube': True,
              'c_tagsLink': 'https://api.census.gov/data/2018/acs/acs5/subject/tags.json',
              'c_valuesLink': 'https://api.census.gov/data/2018/acs/acs5/subject/values.json',
            

In [16]:
df = pd.DataFrame(data = request)

In [17]:
df

@context  \
0  https://project-open-data.cio.gov/v1.1/schema/...   

                                                 @id         @type  \
0  https://api.census.gov/data/2018/acs/acs5/subj...  dcat:Catalog   

                                      conformsTo  \
0  https://project-open-data.cio.gov/v1.1/schema   

                                         describedBy  \
0  https://project-open-data.cio.gov/v1.1/schema/...   

                                             dataset  
0  {'c_vintage': 2018, 'c_dataset': ['acs', 'acs5...

In [30]:
#request['dataset']


'https://api.census.gov/data/2018/acs/acs5/subject/variables.json'

In [31]:
geogapi = request['dataset'][0]['c_geographyLink']
variables = request['dataset'][0]['c_variablesLink']

In [32]:
geogrequest = requests.get(geogapi).json()
varrequest = requests.get(variables).json()

In [44]:
#pprint(geogrequest)
geogrequest["fips"]

for diction in geogrequest["fips"]:
    #print(diction)
    time.sleep(1)
    print( diction["name"])

us
region
division
state
county
county subdivision
subminor civil division
tract
place
consolidated city
alaska native regional corporation
american indian area/alaska native area/hawaiian home land
tribal subdivision/remainder
american indian area/alaska native area (reservation or statistical entity only)
american indian area (off-reservation trust land only)/hawaiian home land
tribal census tract
state (or part)
metropolitan statistical area/micropolitan statistical area
principal city (or part)
metropolitan division
combined statistical area
combined new england city and town area
new england city and town area
principal city
necta division
urban area
congressional district
state legislative district (upper chamber)
state legislative district (lower chamber)
public use microdata area
zip code tabulation area
school district (elementary)
school district (secondary)
school district (unified)


In [46]:
geogrequest["fips"][7]

{'name': 'tract',
 'geoLevelDisplay': '140',
 'referenceDate': '2018-01-01',
 'requires': ['state', 'county'],
 'wildcard': ['county'],
 'optionalWithWCFor': 'county'}

In [47]:
geogapi

'https://api.census.gov/data/2018/acs/acs5/subject/geography.json'

In [49]:
varrequest

{'variables': {'for': {'label': "Census API FIPS 'for' clause",
   'concept': 'Census API Geography Specification',
   'predicateType': 'fips-for',
   'group': 'N/A',
   'limit': 0,
   'predicateOnly': True},
  'in': {'label': "Census API FIPS 'in' clause",
   'concept': 'Census API Geography Specification',
   'predicateType': 'fips-in',
   'group': 'N/A',
   'limit': 0,
   'predicateOnly': True},
  'ucgid': {'label': 'Uniform Census Geography Identifier clause',
   'concept': 'Census API Geography Specification',
   'predicateType': 'ucgid',
   'group': 'N/A',
   'limit': 0,
   'predicateOnly': True},
  'S0804_C04_068E': {'label': 'Estimate!!Public transportation (excluding taxicab)!!Workers 16 years and over who did not work at home!!TIME ARRIVING AT WORK FROM HOME!!5 00 a.m. to 5 29 a.m.',
   'concept': 'MEANS OF TRANSPORTATION TO WORK BY SELECTED CHARACTERISTICS FOR WORKPLACE GEOGRAPHY',
   'predicateType': 'float',
   'group': 'S0804',
   'limit': 0,
   'attributes': 'S0804_C04_0

    {
      "name": "tract",
      "geoLevelDisplay": "140",
      "referenceDate": "2018-01-01",
      "requires": [
        "state",
        "county"
      ],
      "wildcard": [
        "county"
      ],
      "optionalWithWCFor": "county"
    },